src/analysis/schwartz.lean

```diff
@@ -0,0 +1,400 @@
+
+lemma seminorm_aux_nonneg (k n : ℕ) (f : 𝓢(E, F)) : 0 ≤ f.seminorm_aux k n :=
+le_cInf (bounds_nonempty k n f) (λ _ ⟨hx, _⟩, hx)
+
+lemma le_seminorm_aux (k n : ℕ) (f : 𝓢(E, F)) (x : E) :
+  ∥x∥ ^ k * ∥iterated_fderiv ℝ n ⇑f x∥ ≤ f.seminorm_aux k n :=
+le_cInf (bounds_nonempty k n f) (λ y ⟨_, h⟩, h x)
+
+section
+
+open tactic tactic.positivity
+
+/-- Extension for the `positivity` tactic: seminorms are nonnegative. -/
+@[positivity]
+meta def _root_.tactic.positivity_schwartz_seminorm_aux : expr → tactic strictness
+| `(schwartz.seminorm_aux %%a %%b %%c) := nonnegative <$> mk_app ``seminorm_aux_nonneg [a, b, c]
+| _ := failed
+
+end
+
+/-- If one controls the norm of every `A x`, then one controls the norm of `A`. -/
+lemma seminorm_aux_le_bound (k n : ℕ) (f : 𝓢(E, F)) {M : ℝ} (hMp: 0 ≤ M)
+  (hM : ∀ x, ∥x∥^k * ∥iterated_fderiv ℝ n f x∥ ≤ M) :
+  f.seminorm_aux k n ≤ M :=
+cInf_le (bounds_bdd_below k n f) ⟨hMp, hM⟩
+
+lemma seminorm_aux_zero (k n : ℕ) :
+  (0 : 𝓢(E, F)).seminorm_aux k n = 0 :=
+le_antisymm (seminorm_aux_le_bound k n _ rfl.le (λ _, by simp [pi.zero_def])) (by positivity)
+
+lemma seminorm_aux_add_le (k n : ℕ) (f g : 𝓢(E, F)) :
+  (f + g).seminorm_aux k n ≤ f.seminorm_aux k n + g.seminorm_aux k n :=
+(f + g).seminorm_aux_le_bound k n (by positivity) $ λ x, (decay_add_le_aux k n f g x).trans $
+  add_le_add (f.le_seminorm_aux k n x) (g.le_seminorm_aux k n x)
+
+variables [normed_space ℂ F]
+
+lemma seminorm_aux_smul_le (k n : ℕ) (r : ℂ) (f : 𝓢(E, F)) :
+  (r • f).seminorm_aux k n ≤ ∥r∥ * f.seminorm_aux k n :=
+begin
+  refine (r • f).seminorm_aux_le_bound k n (by positivity) (λ x, _),
+  refine (decay_smul_aux' k n f r x).le.trans _,
+  rw mul_assoc,
+  exact mul_le_mul_of_nonneg_left (f.le_seminorm_aux k n x) (norm_nonneg _),
+end
+
+/-- The seminorms of the Schwartz space -/
+@[protected]
+def seminorm (k n : ℕ) : seminorm ℂ 𝓢(E, F) := seminorm.of_smul_le (seminorm_aux k n)
+  (seminorm_aux_zero k n) (seminorm_aux_add_le k n) (seminorm_aux_smul_le k n)
+
+/-- If one controls the norm of every `A x`, then one controls the norm of `A`. -/
+lemma seminorm_le_bound (k n : ℕ) (f : 𝓢(E, F)) {M : ℝ} (hMp: 0 ≤ M)
+  (hM : ∀ x, ∥x∥^k * ∥iterated_fderiv ℝ n f x∥ ≤ M) :
+  seminorm k n f ≤ M := f.seminorm_aux_le_bound k n hMp hM
+
+lemma le_seminorm (k n : ℕ) (f : 𝓢(E, F)) (x : E) :
+  ∥x∥ ^ k * ∥iterated_fderiv ℝ n ⇑f x∥ ≤ seminorm k n f := f.le_seminorm_aux k n x
+
+end seminorms
+
+section topology
+
+/-! ### The topology on the Schwartz space-/
+
+variables (E F)
+variables [normed_space ℂ F]
+
+
+/-- The family of Schwartz seminorms. -/
+def _root_.schwartz_seminorm_family : seminorm_family ℂ 𝓢(E, F) (ℕ × ℕ) :=
+λ n, seminorm n.1 n.2
+
+variables {E F}
+
+instance : topological_space 𝓢(E, F) := (schwartz_seminorm_family E F).module_filter_basis.topology'
+
+instance : has_continuous_smul ℂ 𝓢(E, F) :=
+  (schwartz_seminorm_family E F).module_filter_basis.has_continuous_smul
```

Comment:
```suggestion
(schwartz_seminorm_family E F).module_filter_basis.has_continuous_smul
```
the baseline of proofs is not intendented. Same thing for the next few instances.